manager/integration/tests/test_backing_image.py

```diff
@@ -694,3 +701,190 @@ def test_volume_wait_for_backing_image_condition(client): # NOQA
+    backing_image = backing_image.updateMinNumberOfCopies(minNumberOfCopies=2)
+
+    check_backing_image_disk_map_status(client, BACKING_IMAGE_NAME, 2, "ready")
+
+    update_setting(client, "backing-image-cleanup-wait-interval", "1")
+
+    backing_image = client.by_id_backing_image(BACKING_IMAGE_NAME)
+    backing_image = backing_image.updateMinNumberOfCopies(minNumberOfCopies=1)
+
+    time.sleep(60)
+
+    check_backing_image_disk_map_status(client, BACKING_IMAGE_NAME, 1, "ready")
+    cleanup_all_backing_images(client)
+
+
+@pytest.mark.backing_image  # NOQA
+def test_backing_image_selector_setting(client, volume_name): # NOQA
+    """
+    - Set node1 with nodeTag = [node1], diskTag = [node1]
+    - Set node2 with nodeTag = [node2], diskTag = [node2]
+    - Create a BackingImage with
+        - minNumberOfCopies = 2
+        - nodeSelector = [node1]
+        - diskSelector = [node1]
+    - After creation, the first BackingImage copy will be on node1, disk1
+    - Wait for a while, the second one will never show up
+    - Create the Volume with
+        - numberOfReplicas = 1
+        - nodeSelector = [node2]
+        - diskSelector = [node2]
+    - The volume condition Scheduled will be false
+    """
+
+    tags_1 = ["node1"]
+    tags_2 = ["node2"]
+    nodes = client.list_node()
+    node_1, node_2, node_3 = nodes
+    set_tags_for_node_and_its_disks(client, node_1, tags_1)
+    set_tags_for_node_and_its_disks(client, node_2, tags_2)
+
+    backing_image_node_selector = ["node1"]
+    backing_image_disk_selector = ["node1"]
+    bi_url = BACKING_IMAGE_RAW_URL
+    create_backing_image_with_matching_url(
+        client, BACKING_IMAGE_NAME, bi_url,
+        minNumberOfCopies=2,
+        nodeSelector=backing_image_node_selector,
+        diskSelector=backing_image_disk_selector)
+    time.sleep(10)
+    check_backing_image_disk_map_status(client, BACKING_IMAGE_NAME, 1, "ready")
+
+    volume_node_selector = ["node2"]
+    volume_disk_selector = ["node2"]
+    client.create_volume(name=volume_name, size=str(BACKING_IMAGE_EXT4_SIZE),
+                         numberOfReplicas=1,
+                         diskSelector=volume_disk_selector,
+                         nodeSelector=volume_node_selector,
+                         backingImage=BACKING_IMAGE_NAME)
+    vol = wait_for_volume_detached(client, volume_name)
+    assert vol.diskSelector == volume_disk_selector
+    assert vol.nodeSelector == volume_node_selector
+
+    assert vol.conditions.Scheduled.status == "False"
+    cleanup_all_volumes(client)
+    cleanup_all_backing_images(client)
+
+
+@pytest.mark.backing_image  # NOQA
+def test_backing_image_eviction(client): # NOQA
+    """
+    Test Node Eviction
+    - Create BackingImage with one copy
+    - Evict the node where the copy is on
+    - The BackingImage will be evicted to another node
+
+    Test Disk Eviction
+    - Disable the eviction on the previous node
+    - Evict the disk where the copy is currently on
```

Comment:
It seems that this test:
1. Does not prepare the 2nd disk for the test node
2. always evicts node rather than disks

Reply:
You are right
I change the test to
1. evict the node in the first phase
2. evict the disk in the second phase
    - since each node has only one disk, the backing image will be evicted to other node, and we also check the diskID is different

the reason why we don't prepare another disk is because
=> we create the host disk by getting the host ID of the testing pod and mount the volume to the same node

But when preparing the backing image, we don't know where the backing image is
It may not on the host of the testing pod, thus, we can't prepare the host disk on the different node then the testing pod.
```
def create_host_disk(client, vol_name, size, node_id):
    # create a single replica volume and attach it to node
    volume = create_volume(client, vol_name, size, node_id, 1)

    # prepare the disk in the host filesystem
    disk_path = prepare_host_disk(get_volume_endpoint(volume), volume.name)
    return disk_path

```